rmgpy/molecule/group.py

```diff
@@ -870,3 +918,160 @@ def isIdentical(self, other):
             return False
         else:
             return True
+
+    def standardizeAtomType(self):
+        """
+        This function changes the atomTypes in a group if the atom must
+        be a specific atomType based on its bonds and valency.
+
+        Currently only standardizes oxygen, carbon and sulfur atomTypes
+
+        We also only check when there is exactly one atomType,
+        one bondType, one radical setting.
+        For any group where there are wildcards or multiple attributes,
+        we cannot apply this check.
+
+        In the case where the atomType is ambigious based on bonds
+        and valency, this function will not change the type.
+
+        Returns a 'True' if the group was modified otherwise returns 'False'
+        """
+
+        modified = False
+
+        #dictionary of element to expected valency
+        valency = {atomTypes['C'] : 4,
+                   atomTypes['O'] : 2,
+                   atomTypes ['S']: 2,
+                   atomTypes['Si']:4
+                   }
+
+        #list of :class:AtomType which are elements with more sub-divided atomtypes beneath them
+        specifics= [elementLabel for elementLabel in allElements if not elementLabel in nonSpecifics]
+        for index, atom in enumerate(self.atoms):
+            claimedAtomType = atom.atomType[0]
+            newAtomType = None
+            element = None
+            #Ignore elements that do not have more than one atomtype
+            if claimedAtomType.label in nonSpecifics: continue
+            for elementLabel in specifics:
+                if claimedAtomType.label == elementLabel or atomTypes[claimedAtomType.label] in atomTypes[elementLabel].specific:
+                    element = atomTypes[elementLabel]
+                    break
+
+            #claimedAtomType is not in one of the specified elements
+            if not element: continue
+            #Don't standardize atomtypes for nitrogen for now. My feeling is that
+            # the work on the nitrogen atomtypes is still incomplete
+            elif element == atomTypes['N']: continue
+
+            groupFeatures = getFeatures(atom, atom.bonds)
+            # print groupFeatures
+
+            single = groupFeatures[0]
+            allDouble = groupFeatures[1]
+            triple = groupFeatures[5]
+            benzene = groupFeatures[6]
+
+            bondValency = single + 2 * allDouble + 3 * triple + 4.0/3.0 * benzene
```

Comment:
I'm unsure as well, but I have rewritten if benzene==3, use 4/3 otherwise use 3/2. I think this is the best compromise. 

I thought about making a new bond order 'FB' for fused benzene bond, but even that is not satisfactory because its valency is still unclear. In naphthalene, you might say the two B bonds are 3/2 and FB is 1. However in the case of the central atom in phenalene would you say it has 3 FB bonds with 4/3 each or 2 B bonds and 1 FB bond?